src/org/jitsi/impl/neomedia/rtp/MediaStreamTrackDesc.java

```diff
@@ -0,0 +1,371 @@
+            {
+                isActive = true;
+            }
+
+            rtpEncodings[i].setActive(isActive);
+        }
+
+        RawPacket[] extras = null;
+
+        if (!frameDesc.isInOrder())
+        {
+            // Piggy back till max seen.
+            RawPacketCache inCache = mediaStreamTrackReceiver.getStream()
+                .getCachingTransformer().getIncomingRawPacketCache();
+
+            int start = frameDesc.getStart();
+            int len = (frameDesc.getMaxSeen() - start) & 0xFFFF;
+            extras = new RawPacket[len];
+            for (int i = 1; i <= len; i++)
+            {
+                extras[i] = inCache.get(encoding.getPrimarySSRC(),
+                    (start + i) & 0xFFFF);
+            }
+        }
+        return extras;
+    }
+
+    /**
+     * Finds the {@link RTPEncodingDesc} that corresponds to the packet that is
+     * specified in the buffer passed in as an argument.
+     *
+     * @param buf the byte array that holds the RTP packet.
+     * @param off the offset in the byte array where the actual data starts
+     * @param len the length of the actual data
+     * @return the {@link RTPEncodingDesc} that corresponds to the packet that is
+     * specified in the buffer passed in as an argument, or null.
+     */
+    public RTPEncodingDesc findRTPEncodingDesc(byte[] buf, int off, int len)
+    {
+        if (buf == null || buf.length < off + len)
+        {
+            return null;
+        }
+
+        for (RTPEncodingDesc encoding : rtpEncodings)
+        {
+            if (encoding.matches(buf, off, len))
+            {
+                return encoding;
+            }
+        }
+
+        return null;
+    }
+
+    /**
+     * Finds the {@link FrameDesc} that corresponds to the packet that is
+     * specified in the buffer passed in as an argument.
+     *
+     * @param buf the byte array that holds the RTP packet.
+     * @param off the offset in the byte array where the actual data starts
+     * @param len the length of the actual data
+     * @return the {@link FrameDesc} that corresponds to the packet that is
+     * specified in the buffer passed in as an argument, or null.
+     */
+    public FrameDesc findFrameDesc(byte[] buf, int off, int len)
+    {
+        RTPEncodingDesc rtpEncoding = findRTPEncodingDesc(buf, off, len);
+        if (rtpEncoding == null)
+        {
+            return null;
+        }
+
+        return rtpEncoding.findFrameDesc(buf, off, len);
+    }
+
+    /**
+     * Makes a receiver report for this track.
+     *
+     * @param nowMs
+     * @return
+     */
+    public List<RTCPReportBlock> makeReceiverReport(long nowMs)
+    {
+        SSRCCache cache = mediaStreamTrackReceiver
+            .getStream().getRTPTranslator().getSSRCCache();
+
+        List<RTCPReportBlock> reportBlocks
+            = new ArrayList<>(rtpEncodings.length);
+
+        if (cache != null)
+        {
+            for (int i = 0; i < rtpEncodings.length; i++)
```

Comment:
Wouldn't this create multiple report blocks for each SSRC for which temporal scalability is configured (i.e. which belongs to more than one RTP Encoding)?